### PR TITLE
Exported `unsupportedMediaType`.

### PR DESCRIPTION
**Pull request by taphu**
See commits for comments.

### CODE REVIEW

File: src/Web/Moonshine.hs [hunk 1]
```diff
@@ -468,7 +469,13 @@ readEntity maxSize = do
...
   where
     getRequestContentType = getsRequest (getHeader "content-type")
-    unsupportedMediaType = finishWith . setResponseCode 415 =<< getResponse
+
+
+{- |
+  Short circuit with a 415 response.
+-}
+unsupportedMediaType :: Snap a
+unsupportedMediaType = finishWith . setResponseCode 415 =<< getResponse
```
Comment (potomak) [inline src/Web/Moonshine.hs]: Why did you change the usual:

```haskell
do
  response <- fmap (setResponseCode XXX) getResponse
  finishWith response
```

Comment (potomak) [inline src/Web/Moonshine.hs]:
```haskell
respondWithStatusCode :: Int -> Snap a
respondWithStatusCode statusCode = finishWith . setResponseCode statusCode =<< getResponse
```

Comment (taphu) [inline src/Web/Moonshine.hs]: No reason.

Comment (taphu) [inline src/Web/Moonshine.hs]: They are the same, and I think I like the new one better. But I will make them all the same if you think that's important.

Comment (potomak) [inline src/Web/Moonshine.hs]: Just merged. I asked for the sake of consistency, but it was a minor issue.
